bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/http/RateLimitedHttpClient.java

```diff
@@ -112,18 +210,140 @@ private void stopProcessJob() {
 
     private void processQueue() {
         RequestQueueEntry queueEntry = requestQueue.poll();
-        if (queueEntry != null) {
-            queueEntry.future.complete(httpClient.newRequest(queueEntry.finalUrl));
+        execute(queueEntry);
+    }
+
+    private HttpClient getHttpClient() {
+        return Optional.ofNullable(this.httpClient)
```

Comment:
Creating an optional for null-checking is considered bad style.

Reply:
Ok, i'm new to Java. How can i translate a @Nullable to be accepted by a @nonnull? 
this was the only way i found.

Reply:
Why is this `@Nullable` in the first place? That's essentially the same question I had below: Why did you remove the client assignment from the constructor?

You could return the `Optional` and use `isPresent` or `ifPresent` in the caller or you could assign the field to a local variable and check that against null.

Reply:
I removed it from the ctor because in the execution flow it is never used. 
First use of the httpClient is after initialize is called.

So i thought it is better to instantiate in the initialize and not in the ctor to throw it away in the initialize.